src/components/Transaction.jsx

```diff
@@ -0,0 +1,85 @@
+import React from 'react'
+import { useEffect } from 'react';
+import { useState } from 'react'
+import { getAllCheckingAccounts, transaction } from '../api';
+
+export const Transaction = (props) => {
+
+
+    const [accounts,setAccounts] = useState([]); 
+    const [loading,setLoading] = useState(true); 
+    const [transactionValues,setTransactionValues] = useState({
+        amount:'',
+        currentAccount:''
+    }); 
+    const [error,setError] = useState(false); 
+    const [success,setSuccess] = useState(false); 
+
+    useEffect(()=>{
+        getAllCheckingAccounts()
+        .then(response =>{
+            setAccounts(response.data);
+            setLoading(false);  
+        },[]); 
+    })
+
+    const handleInputChange = ({target})=>{
+        
+        setTransactionValues({
+            ...transactionValues,
+            [target.name]:target.value
+        }) 
+    }
+
+    const handleSubmit = async(e) =>{
+        e.preventDefault(); 
+        try{
+            await transaction(props.type,transactionValues.amount,props.type,transactionValues.currentAccount);
```

Comment:
Is there a reason why you pass the `type` twice?

Also, I'd add the `type` to the transaction object and pass that as a parameter to this function.

Reply:
I'm passing the `type` twice because in that specific endpoint the type is required and also the description, so I'm sending the type as transaction description.

Reply:
But I think I can put all that data in an object and send it as parameter as you said